File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C/s936984238_comments_removed.c
#include <stdio.h>
#include <stdlib.h>

int *score;
int number;

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int read() {
    if (score != NULL) {
        free(score);
        score = NULL;
    }
    if (scanf("%d", &number) != 1) return 0;
    if (number == 0) return 0;
    score = malloc(number * sizeof(int));
    for (int i = 0; i < number; i++) {
        scanf("%d", &score[i]);
    }
    return 1;
}

void solve() {
    qsort(score, number, sizeof(int), compare);
    int sum = 0;
    for (int i = 1; i < number - 1; i++) {
        sum += score[i];
    }
    printf("%d\n", sum / (number - 2));
}

int main() {
    while (read()) {
        solve();
    }
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/Go/s936984238_comments_removed.go
package main

import (
	"fmt"
	"sort"
)

var score []int
var number int

func main() {
	for read() {
		solve()
	}
}

func read() bool {
	_, err := fmt.Scanf("%d", &number)
	if err != nil || number == 0 {
		return false
	}

	score = make([]int, number)
	for i := 0; i < number; i++ {
		fmt.Scanf("%d", &score[i])
	}
	return true
}

func solve() {
	sort.Ints(score)
	sum := 0
	for i := 1; i < number-1; i++ {
		sum += score[i]
	}
	fmt.Println(sum / (number - 2))
}
// End of Code
